src/Analyzers/CSharp/Tests/UseParameterNullChecking/UseParameterNullCheckingTests.cs

```diff
@@ -452,7 +452,14 @@ public C(string s!!)
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
         public async Task TestRedundantNullChecks()
         {
-            // TODO: what test framework settings do we use here?
+            // Produces the following error:
+            //             Message: 
+            // System.InvalidOperationException : GetCurrentNode returned null with the following node: string s - line 321
+            //  Stack Trace: 
+            // Contract.Fail(String message, Int32 lineNumber) line 115
+            // ReplaceChange.Apply(SyntaxNode root, SyntaxGenerator generator) line 321
+            // SyntaxEditor.GetChangedRoot() line 92
+            // <FixAllWithEditorAsync>d__5.MoveNext() line 71
```

Comment:
https://github.com/dotnet/roslyn/issues/41526 https://github.com/dotnet/roslyn/issues/53960 👀

Reply:
I think the other issues may be a bit different, and I think I got what's happening.

For this scenario, we produced two diagnostics, with the same parameter. The fixer removed the first node and replaced the ParameterSyntax node. Then, it attempts to remove the second node and... when it tries to replace the ParameterSyntax node it doesn't find it. It was already replaced.

```
                var parameterSyntax = (ParameterSyntax)diagnostic.AdditionalLocations[0].FindNode(cancellationToken);
```

We need a way to only try to get this `parameterSyntax` if it wasn't already replaced. Or, consolidate diagnostics related to the same parameter into one diagnostic, for example with location as the first node to be removed, and `AdditionalLocations[0]` as the parameter to be replaced, and remaining `AdditionalLocations` as other nodes to be removed.

Reply:
> We need a way to only try to get this `parameterSyntax` if it wasn't already replaced.

I was thinking of just doing this with a `PooledHashSet<SyntaxNode>`.

> Or, consolidate diagnostics related to the same parameter into one diagnostic

This also seems doable, but maybe cumbersome on the analyzer side? Also, it means the code fix experience will be to "remove all null checks for the parameter"--versus, say, use the fixer to handle this one null check, then notice that the check later on in the code is erroneous and make an edit manually. Perhaps the parameter was reassigned and then null-tested, and we were throwing `ArgumentNullException` when we maybe should have been throwing a different kind of exception.

Reply:
Have resolved by adding a set of locations and avoiding replacing the node at the same location multiple times.